product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -476,6 +476,7 @@ private void updateProductRelations(ProductPutVm productPutVm, Product product)
 
     private List<ProductOptionValue> updateProductOptionValues(ProductPutVm productPutVm, Product product,
                                                             Map<Long, ProductOption> optionsById) {
+        productOptionValueRepository.deleteAllByProductId(product.getId());
```

Comment:
Hi @NhatTranMinh15,
I would suggest using Cascade with orphan removal to manage the consistency of product and product option values automatically and reduce boilerplate code.
Could you give it a try?

Reply:
I've tried it. But the product and product option value are independent of each other, only product option value hold the @ManyToOne relationship with product, so I can't use cascade within product.
In the context of this PR, the deletion of product option value is to remove the duplication values inside the database.

Reply:
Hi @NhatTranMinh15,
> But the product and product option value are independent of each other

I think it should not be like that. We can't have ProductOptionValue without a Product (the ProductOptionValue lifecycle is entirely dependent on Product). 
What I suggest is adding the following relationship mapping to Product:

`@OneToMany(mappedBy = "product", cascade = {CascadeType.ALL}, orphanRemoval = true)
    private List<ProductOptionValue> productOptionValues = new ArrayList<>();`
    
By this way, we don't need to handle the insertion/removal of productOptionValue manually. It seems we need to refactor a little bit in other parts.

Reply:
Let's me try this to see if it works